README.md

```diff
@@ -125,13 +126,65 @@ const exampleBlockedTiersInDaysData = {
     {
       charge: -0.08633,
       min: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] // 12 months,
-      max: [12, 12, 12, 12, 14.2, 14.2, 14.2, 14.2, 14.2, 14.2, 12, 12] // 12 months,
+      max: [12, 12, 12, 12, 14.2, 14.2, 14.2, 14.2, 14.2, 14.2, 12, 12], // 12 months,
       name: 'Discount',
     },
   ],
 }
 ```
 
+**SurchargeAsPercent**: Used to specify surcharges such as taxes or other percent based charges. For example, some rates have "improvement" charges which are calculated based on a subset of the other rate elements.
+
+Unlike other rate elements, the `SurchargeAsPercent` does not explicitly define `rateComponents`.
+
+Instead, it will dynamically create `RateComponent`s that calculate a charge based on the specified percentage for each of the rate elements defined by its `rateElementIds` property. If the `appliesToAll` property is set to true, then `RateComponents` will be created for every rate element.
+
+```js
+const exampleWithSurcharges = [
+  {
+    rateElementType: 'SurchargeAsPercent',
+    name: 'Sales tax',
+    percent: 7.25, // note: not defined as a decimal,
+    appliesToAll: true,
```

Comment:
I'd prefer to avoid having an API where some options can negate other options. Can we have `rateElementIds` control this entire concept? If rateElementIds are provided, we restrict to those, but if they aren't, we assume it should be applied to all? I believe this would mimic how we treat filters (providing them filters, omitting them allows all)